### PR TITLE
added default headers to constants

### PR DESCRIPTION
**Pull request by doubledare704**
passing new parameter to Socket init
converting dict of headers into Headers object while connecting to ws with old client in websockets.

## What kind of change does this PR introduce?

Feature

## What is the current behavior?

Socket class connects with default headers

## What is the new behavior?

Adding custom headers into _connect method as new optional parameter in class

## Additional context

Issue from [link](https://github.com/supabase/realtime-py/issues/15)

### CODE REVIEW

File: realtime_py/constants.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+__all__ = [
+    "DEFAULT_HEADERS"
+]
+
+from .version import __version__
+
+DEFAULT_HEADERS = {
+    "X-Client-Info": f"gotrue-py/{__version__}"
```
Comment (awalias) [inline realtime_py/constants.py:8]: I think this needs to be `realtime-py`

File: realtime_py/version.py [hunk 2]
```diff
@@ -0,0 +1 @@
+__version__ = "0.1.0"
```
Comment (awalias) [inline realtime_py/version.py:1]: is this one static or does it get overwritten dynamically anywhere?

**Review by awalias — CHANGES_REQUESTED** [review]

**Comment by awalias** [comment]
thanks for picking this one up btw @doubledare704 ! I actually forgot that we can't send headers via the ws connection

**Comment by dreinon** [comment]
@awalias then should we close this PR?

**Comment by J0** [comment]
Hey @doubledare704,

Thanks for your contribution! Unfortunately, it seems like we have to close this for now since we can't send headers over websockets. We really appreciate the time and effort put into this PR though! We can close this for now if we find a workaround.

 If you'd like some stickers do feel free to reach out to me on Discord(J0).

@dreinon @awalias feel free to chime in if you have any concerns/thoughts as well.

Thanks!
Joel

**Pull request closed without merging**